File: backend/src/main/java/com/backend/domain/community/dto/response/CommunityResponseDTO.java
package com.backend.domain.community.dto.response;

import com.backend.domain.analysis.entity.AnalysisResult;
import com.backend.domain.analysis.entity.Score;
import com.backend.domain.repository.entity.Repositories;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

public record CommunityResponseDTO(
        String userName,
        String userImage,
        String repositoryName,
        Long repositoryId,
        String summary,
        String description,
        List<String> language,
        int totalScore,
        LocalDateTime createDate,
        boolean publicStatus,
        String htmlUrl
) {
    public CommunityResponseDTO(Repositories repositories, AnalysisResult analysis, Score score) {
        this(
                repositories.getUser().getName(),
                repositories.getUser().getImageUrl(),
                repositories.getName(),
                repositories.getId(),
                analysis.getSummary(),
                repositories.getDescription(),
                repositories.getLanguages().stream()
                        .map(language -> language.getLanguage().name())
                        .collect(Collectors.toList()),
                score.getTotalScore(),
                analysis.getCreateDate(),
                repositories.isPublicRepository(),
                repositories.getHtmlUrl()
        );
    }
}

File: backend/src/main/java/com/backend/domain/community/dto/response/CommunityResponseDTO.kt
import com.backend.domain.analysis.entity.AnalysisResult
import com.backend.domain.analysis.entity.Score
import com.backend.domain.repository.entity.Repositories
import java.time.LocalDateTime

data class CommunityResponseDTO(
    val userName: String,
    val userImage: String?,
    val repositoryName: String,
    val repositoryId: Long,
    val summary: String,
    val description: String?,
    val language: MutableList<String>,
    val totalScore: Int,
    val createDate: LocalDateTime,
    val publicStatus: Boolean,
    val htmlUrl: String
) {
    constructor(repositories: Repositories, analysis: AnalysisResult, score: Score) : this(

        // 프로퍼티 접근으로 수정
        userName = repositories.user.name,
        userImage = repositories.user.imageUrl,

        repositoryName = repositories.name,
        repositoryId = repositories.id!!,

        summary = analysis.summary,
        description = repositories.description,

        language = repositories.languages
            .map { it.language.name }
            .toMutableList(),

        totalScore = score.totalScore,
        createDate = analysis.createDate,

        publicStatus = repositories.isPublic,
        htmlUrl = repositories.htmlUrl
    )
}
